libs/ngx-charts-on-fhir/src/lib/fhir-chart/fhir-chart.component.ts

```diff
@@ -71,8 +72,15 @@ export class FhirChartComponent implements OnInit {
 
     // we use a custom legend component instead
     Chart.defaults.plugins.legend.display = false;
+    this.getChartConfiguration();
+    this.configService.annotationSubject.subscribe((timeframeAnnotations: ChartAnnotation[]) => {
+      this.configService.updateChartConfiguration(timeframeAnnotations);
+      this.getChartConfiguration();
+    });
+  }
 
-    this.configService.chartConfig$.subscribe((config) => {
+  getChartConfiguration() {
+    this.configService.chartConfig$?.subscribe((config) => {
       this.datasets = config.data.datasets;
       this.options = config.options;
     });
```

Comment:
Why is this component listening to annotationSubject? the component doesn't use the annotations. that should be an internal implementation detail of the service.

The service shouldn't need another component to tell it when to update its configuration. The service should be listening to the necessary events itself and updating its own configuration when needed.